core/logger/logger.go

```diff
@@ -192,42 +196,80 @@ func NewLogger() Logger {
 		parseErrs = append(parseErrs, invalid)
 	}
 
-	c.ToDisk, invalid = envvar.LogToDisk.ParseBool()
+	var fileMaxSize utils.FileSize
+	fileMaxSize, invalid = envvar.LogFileMaxSize.ParseFileSize()
+	c.DiskMaxSizeBeforeRotate = int(fileMaxSize)
 	if invalid != "" {
 		parseErrs = append(parseErrs, invalid)
 	}
 
+	if c.DebugLogsToDisk() {
+		var (
+			fileMaxAge int64
+			maxBackups int64
+		)
+
+		fileMaxAge, invalid = envvar.LogFileMaxAge.ParseInt64()
+		c.DiskMaxAgeBeforeDelete = int(fileMaxAge)
+		if invalid != "" {
+			parseErrs = append(parseErrs, invalid)
+		}
+
+		maxBackups, invalid = envvar.LogFileMaxBackups.ParseInt64()
+		c.DiskMaxBackupsBeforeDelete = int(maxBackups)
+		if invalid != "" {
+			parseErrs = append(parseErrs, invalid)
+		}
+	}
+
 	c.UnixTS, invalid = envvar.LogUnixTS.ParseBool()
 	if invalid != "" {
 		parseErrs = append(parseErrs, invalid)
 	}
 
-	l := c.New()
+	l, close := c.New()
 	for _, msg := range parseErrs {
 		l.Error(msg)
 	}
-	return l.Named(verShaNameStatic())
+	return l.Named(verShaNameStatic()), close
 }
 
 type Config struct {
-	LogLevel    zapcore.Level
-	Dir         string
-	JsonConsole bool
-	ToDisk      bool // if false, the Logger will only log to stdout.
-	UnixTS      bool
+	LogLevel                   zapcore.Level
+	Dir                        string
+	JsonConsole                bool
+	UnixTS                     bool
+	DiskMaxSizeBeforeRotate    int // megabytes
+	DiskMaxAgeBeforeDelete     int // days
+	DiskMaxBackupsBeforeDelete int // files
```

Comment:
Would these still be clear without the suffixes?
```suggestion
	DiskMaxSize    int // megabytes
	DiskMaxAge     int // days
	DiskMaxBackups int // files
```
Or like this?
```suggestion
	DiskSizeMax    int // megabytes
	DiskAgeMax     int // days
	DiskBackupsMax int // files
```
And are `int`s the best units for size and age? Or could we use `utils.FileSize` and `time.Duration`?

Reply:
```suggestion
	FileSizeMax    utils.FileSize
	FileAgeMax     time.Duration
	FileBackupsMax int 
```

Reply:
`int` is being used here to match Lumberjack's typing. it is expecting `int` for all those values.